add back cobertura

tpapp opened the pull request:
Reverting parts of #30.

tpapp commented:
@gabriel-ss, I missed the fact that #30 removes Covertura functionality, so I reverted that part. Would you please review?

@bramtayl: reading the code I realized that we don't use `PYTHON` anywhere, so I removed it. Would you please test with `lcov-cobertura` on your setup?

gabriel-ss commented:
@tpapp Sorry for the trouble, generate_xml wasn't on the RFC, so I removed the wrapper. The changes look fine to me, tested the xml functionality, and it's working here.

bramtayl commented:
Nope, the tests don't pass for me as is, but the tests below do. The table line regular expression wasn't working, because the forward slash was replaced by a double slash, and this pushed the l at the end to the next line. Also, I disabled the the genhtml test because lcov only seems to be available on linux. PS why not build lcov with BinaryBuilder?

```
using LocalCoverage, Test

table_header = r"File name\s+.\s+Lines hit\s+.\s+Coverage\s+.\s+Missing"
table_line = r"src(\/|\\\\)LocalCoverage.jl?\s+.\s+\d+\s*\/\s*\d+\s+.\s+\d+%\s+.\s+"
table_footer = r"TOTAL\s+.\s+\d+\s*\/\s*\d+\s+.\s+\d+%\s+.\s+."

lockfile = joinpath(tempdir(), "testingLocalCoverage") # prevent infinite recursion when testing

covdir = normpath(joinpath(@__DIR__, "..", "coverage"))

if !isfile(lockfile)
    @test isdir(LocalCoverage.pkgdir("LocalCoverage"))
    tracefile = joinpath(covdir, "lcov.info")
    @test !isfile(tracefile)
    try
        touch(lockfile)

        cov = generate_coverage("LocalCoverage")
        buffer = IOBuffer()
        show(buffer, cov)
        table = String(take!(buffer))
        @test !isnothing(match(table_header, table))
        @test !isnothing(match(table_line, table))
        @test !isnothing(match(table_footer, table))

        if Sys.islinux()
            mktempdir() do dir
                html_coverage("LocalCoverage", dir = dir)
                @test isfile(joinpath(dir, "index.html"))
            end
        end
        @test isfile(tracefile)
    finally
        rm(lockfile)
        rm(covdir, recursive = true)
    end

    @test LocalCoverage.find_gaps([nothing, 0, 0, 0, 2, 3, 0, nothing, 0, 3, 0, 6, 2]) ==
          [2:4, 7:7, 9:9, 11:11]
end
```

gabriel-ss commented:
> Nope, the tests don't pass for me as is, but the tests below do. The table line regular expression wasn't working, because the forward slash was replaced by a double slash, and this pushed the l at the end to the next line. Also, I disabled the the genhtml test because lcov only seems to be available on linux. PS why not build lcov with BinaryBuilder?
> 
> ```
> using LocalCoverage, Test
> 
> table_header = r"File name\s+.\s+Lines hit\s+.\s+Coverage\s+.\s+Missing"
> table_line = r"src(\/|\\\\)LocalCoverage.jl?\s+.\s+\d+\s*\/\s*\d+\s+.\s+\d+%\s+.\s+"
> table_footer = r"TOTAL\s+.\s+\d+\s*\/\s*\d+\s+.\s+\d+%\s+.\s+."
> 
> lockfile = joinpath(tempdir(), "testingLocalCoverage") # prevent infinite recursion when testing
> 
> covdir = normpath(joinpath(@__DIR__, "..", "coverage"))
> 
> if !isfile(lockfile)
>     @test isdir(LocalCoverage.pkgdir("LocalCoverage"))
>     tracefile = joinpath(covdir, "lcov.info")
>     @test !isfile(tracefile)
>     try
>         touch(lockfile)
> 
>         cov = generate_coverage("LocalCoverage")
>         buffer = IOBuffer()
>         show(buffer, cov)
>         table = String(take!(buffer))
>         @test !isnothing(match(table_header, table))
>         @test !isnothing(match(table_line, table))
>         @test !isnothing(match(table_footer, table))
> 
>         if Sys.islinux()
>             mktempdir() do dir
>                 html_coverage("LocalCoverage", dir = dir)
>                 @test isfile(joinpath(dir, "index.html"))
>             end
>         end
>         @test isfile(tracefile)
>     finally
>         rm(lockfile)
>         rm(covdir, recursive = true)
>     end
> 
>     @test LocalCoverage.find_gaps([nothing, 0, 0, 0, 2, 3, 0, nothing, 0, 3, 0, 6, 2]) ==
>           [2:4, 7:7, 9:9, 11:11]
> end
> ```

That's definitively more robust, I think we could just do `if !isnothing(Sys.which("genhtml"))` instead of  `if Sys.islinux()` to cover the broader case where html_coverage isn't used.

gabriel-ss commented on src/LocalCoverage.jl:
```diff
@@ -195,8 +190,7 @@ end
 
-Generate, and optionally open, the HTML coverage summary in a browser for `pkg`
-inside `dir`.
+Open the HTML coverage results in a browser for `pkg` if they exist.
```
I think juet that we could keep the information about the other parameters in this doc.
```suggestion
Generate, and optionally open, the HTML coverage summary in a browser for `pkg`
inside `dir`.
```

tpapp commented:
@bramtayl: thanks for the suggestions, I incorporated them.

@gabriel-ss: I added back `clean_coverage` and made a few minor changes to your code. It would be great if you could review the last 2 commits, then I would merge and release.

gabriel-ss reviewed (approved):
Looks great! Sorry once again for the removed functions.
